### ISSUE TITLE
input coercion errors should be ValidationErrors

### ISSUE THREAD

**Issue by schmod** [apollographql/apollo-server#3304]
If I make a request that is illegal because the input arguments/variables are the wrong type, I would expect Apollo to emit a `ValidationError`.  Instead, the more generic `GraphQLError` is thrown.

A very naive example:

This Query:
```graphql
query UserQuery($userId: Int!) {
  user(userId: $userId) {
    firstName
  }
}
```

with the variables:
```json
{ "userId": null }
```

produces an error object (passed to `formatError`) that looks something like:

```json
{
  "message":  "Variable \"$userId\" of non-null type \"Int!\" must not be null.",
  "name":"GraphQLError"
}
```

---

Because these errors are typically the result of a bad client request, I would like to have some sort of mechanism where I can ensure that they are logged with a lower severity (compared to an actual execution error).

**Comment by schmod** [comment]
The coercion process is called by [graphql/execution/values.js](https://github.com/graphql/graphql-js/blob/master/src/execution/values.js#L73)
which is in turn called by [graphql/execution/execute](https://github.com/graphql/graphql-js/blob/master/src/execution/execute.js#L319)

I wasn't able to locate a good point where Apollo could unambiguously intercept these errors 😕 – I wonder if this is a change that would need to be made in graphql-js itself....

**Comment by n1ru4l** [comment]
@schmod I mentioned a possible (but really ugly) fix here: https://github.com/apollographql/apollo-server/issues/3498#issuecomment-554966929

Is there currently any discussion ongoing on changing this behaviour you mentioned? Would this involve changing the GraphQL spec or is this something that must be handled inside the implementations (such as graphql-js)?

**Comment by glasser** [comment]
This is a duplicate of #5353, I think. Note that we fixed some of this in #5091.
